### PR TITLE
Clarify that it is ok to use plt.subplots to create figures

### PR DESCRIPTION
**Pull request by ianhi**
## PR Summary
Adds a line explaining that it is ok to use pyplot methods to create figures. This was (is perhaps) confusing for me and seems to be for other users as well https://discourse.matplotlib.org/t/how-do-i-use-the-object-oriented-interface-without-having-to-rely-on-pyplot/21877

cc @jklymak because this is based on your response to that discourse post.

## PR Checklist

- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: tutorials/introductory/lifecycle.py [hunk 1]
```diff
@@ -41,7 +41,8 @@
 .. note::
 
    In general, try to use the object-oriented interface over the pyplot
```
Comment (jklymak) [inline tutorials/introductory/lifecycle.py:43]:
```suggestion
   In general, the explicit interface is preferred over the pyplot
```

File: tutorials/introductory/lifecycle.py [hunk 2]
```diff
@@ -41,7 +41,8 @@
...
-   interface.
+   interface for making the plots. However, for creating figures and axes
+   the pyplot interface with `~.pyplot.subplots` should be preferred.
```
Comment (jklymak) [inline tutorials/introductory/lifecycle.py:45]:
```suggestion
   interface for making the plots. The `~.pyplot` module still has some useful 
   shortcuts, and in particular is the preferable method for instantiating figures
   and axes via ``fig=plt.figure()`` or ``fig, axs=plt.subplots(2, 1)``. 
```

**Comment by jklymak** [comment]
Maybe something like the above?  Or even more detail.  There is lots of good stuff in pyplot that this note was not meant to forbid. ie. plt.rcParams is pretty helpful, rather than importing matplotlib.

File: tutorials/introductory/lifecycle.py (hunk 1 above)
Comment (ianhi) [reply tutorials/introductory/lifecycle.py:43]: That would require rephrasing the earlier paragraphs in the tutorial as well. That may be a good thing to do but wasn't my intention here. Is this enough of an incremental improvement to stand on its own?

File: tutorials/introductory/lifecycle.py (hunk 2 above)
Comment (ianhi) [reply tutorials/introductory/lifecycle.py:45]: I like this.

File: tutorials/introductory/lifecycle.py (hunk 1 above)
Comment (jklymak) [reply tutorials/introductory/lifecycle.py:43]: `explicit object-oriented..`?

Comment (ianhi) [reply tutorials/introductory/lifecycle.py:43]: That's better but it is still a break from how the opening paragraph describes things. But perhaps Im worrying too much about that

Comment (jklymak) [reply tutorials/introductory/lifecycle.py:43]: Maybe fix the first parapraph while we are here?  It is not terribly clear.

**Comment by jklymak** [comment]
@ianhi are you coming back to this?

**Comment by rcomer** [comment]
This section had a bit of a re-write at ##22613 and the relevant note now says "...for plotting".  I there more to do here?

**Comment by timhoffm** [comment]
Wecshiuld align with #26388, which is sort of orthogonal but would need a rewrite of the same sections. I‘d be ok with closing here.

**Pull request closed without merging**